cartographer_ros_msgs/srv/TrajectoryQuery.srv

```diff
@@ -0,0 +1,20 @@
+# Copyright 2019 The Cartographer Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+int32 trajectory_id
+time min_time
+time max_time
+---
+cartographer_ros_msgs/StatusResponse status
+geometry_msgs/PoseStamped[] trajectory
```

Comment:
Using `geometry_msgs/PoseStamped` relative to the global map frame. In case of a deserialized trajectory, I think we can't recover the ROS tracking frame (no child frame ID)?

Reply:
Remember my rgf about the API for storing user data in the proto stream? :-)